Fill.py

```diff
@@ -502,7 +502,12 @@ def mark_for_locking(location: Location):
         # "priority fill"
         fill_restrictive(multiworld, multiworld.state, prioritylocations, progitempool,
                          single_player_placement=single_player, swap=False, on_place=mark_for_locking,
-                         name="Priority", one_item_per_player=False)
+                         name="Priority", one_item_per_player=True, allow_partial=True)
+
+        # retry with one_item_per_player off because some priority fills can fail to fill with that optimization
+        fill_restrictive(multiworld, multiworld.state, prioritylocations, progitempool,
+                         single_player_placement=single_player, swap=False, on_place=mark_for_locking,
+                         name="Priority Retry", one_item_per_player=False)
```

Comment:
This should probably be blocked inside a `prioritylocations` check so that the fill code isn't run unnecessarily
```suggestion
        if prioritylocations:
            # retry with one_item_per_player off because some priority fills can fail to fill with that optimization
            fill_restrictive(multiworld, multiworld.state, prioritylocations, progitempool,
                            single_player_placement=single_player, swap=False, on_place=mark_for_locking,
                            name="Priority Retry", one_item_per_player=False)
```

Reply:
I agree with this, Medic and I discussed this after they had a look at the code.
The behavior is correct, the retry `fill_restrictive` will do nothing if the first call placed everything successfully.
However, the amount of code it runs to "do nothing" is enough to justify checking again & not calling it at all if it's not necessary.